### PR TITLE
parse/size: support IEEE 1541 size units v2

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1971

https://redmine.openinfosecfoundation.org/issues/1457

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21483

File: suricata.yaml.in [hunk 1]
```diff
@@ -410,7 +410,7 @@ outputs:
 
       # File size limit.  Can be specified in kb, mb, gb.  Just a number
```
Comment (inashivb) [inline suricata.yaml.in:411]: nit: comment still says kb, mb, gb..

File: suricata.yaml.in [hunk 2]
```diff
@@ -314,8 +314,8 @@ outputs:
         - ssh
         - mqtt:
             # passwords: yes           # enable output of passwords
-            # string-log-limit: 1kb    # limit size of logged strings in bytes.
-                                       # Can be specified in kb, mb, gb. Just a number
+            # string-log-limit: 1KiB   # limit size of logged strings in bytes.
+                                       # Can be specified in KiB, MiB, GiB. Just a number
                                        # is parsed as bytes. Default is 1KB.
```
Comment (inashivb) [inline suricata.yaml.in:319]: nit: KB

**Review by inashivb — COMMENTED** [review]
Looking good. Some nits inline and missing small updates to the configuration file below.

Config options:
- `types.alert.payload-buffer-size: 4kb`
- `types.frame.payload-buffer-size: 4kb`
- `datasets.defaults.memcap: 100mb`

Config comments:
- `the reserved memory (memcap) for flows is 32MB`
- `The memcap can be specified in kb, mb, gb`
- `reassemble 1mb into a stream `
- `Default is unlimited, values can be 64mb`

**Comment by victorjulien** [comment]
Replaced by #11485

**Pull request closed without merging**